### PR TITLE
Yaroslav karabin

### PR DESCRIPTION
**Pull request by Kavjar**
Already done this task ;)

### CODE REVIEW

**Review by BodiaPavlyk — APPROVED** [review]

File: unittest/test_algo.py [hunk 1]
```diff
@@ -135,4 +135,81 @@ class TestTask331b(unittest.TestCase):
...
+        (12, 36, [36, 72, 108, 144, 180, 216, 252, 288, 324, 360, 396]), (78, 5, []),
+        (78, 12, [156, 312, 468, 624, 780]),
+        (444, 822, [60828, 121656, 182484, 243312, 304140]), (77, 127, [])])
+    def test_main_logic(self, number1, number2, expected_value):
+        self.assertEqual(algo.Task226.main_logic(number1, number2), expected_value)
+
+
+class TestTask559(unittest.TestCase):
+    # Testing base class with validation method for exceptions
+    @parameterized.expand([('1.2', TypeError), ('lol', TypeError),
```
[the remark this answers is not in the document]
Comment (Kavjar) [reply unittest/test_algo.py:230]: totally agree with you

File: pytest/test_algo.py [hunk 2]
```diff
@@ -117,4 +117,81 @@ def test_task331a_main_logic(number, expected_value):
...
 def test_task331b_main_logic(number, expected_value):
-    assert algo.Task331b.main_logic(number) == expected_value
\ No newline at end of file
+    assert algo.Task331b.main_logic(number) == expected_value
+
+
+# Testing base class with validation method for exceptions
+@pytest.mark.parametrize('data, expected_value', [('1.2', ValueError), ('-3ssss', ValueError), ('0', ValueError),
+                                                  ('445', ValueError), ('7m 456', TypeError), ('7115m 4', TypeError),
+                                                  ('124 4d56', TypeError), ('7,5555 456', TypeError),
```
Comment (BurlyLlama28) [inline pytest/test_algo.py:171]: Here I see this trouble too, equal lines would me more readable

Comment (Kavjar) [reply pytest/test_algo.py:171]: ok

**Review by BurlyLlama28 — COMMENTED** [review]
I liked rearrangement of code, well done:)

**Pull request merged**